### PR TITLE
opengl:  return 0 when make current success in iOS EAGL

### PR DESCRIPTION
**Pull request by hsong-gpsw**
This eliminated false error message of "Failed resize video engine" in video engine

### CODE REVIEW

File: libnodegl/src/backends/gl/glcontext_eagl.m [hunk 1]
```diff
@@ -293,7 +293,7 @@ static int eagl_make_current(struct glcontext *ctx, int current)
         ret = [EAGLContext setCurrentContext:nil];
     }
 
-    return ret;
+    return ret == 1? 0: -1;
```
Comment (rlavarec-gpsw) [inline libnodegl/src/backends/gl/glcontext_eagl.m:296]: https://developer.apple.com/documentation/opengles/eaglcontext/1624882-setcurrentcontext
Seems to return a bool.
We could do != FALSE to rule out TRUE not been 1.
` ret != FALSE? 0: -1;`

Comment (hsong-gpsw) [reply libnodegl/src/backends/gl/glcontext_eagl.m]: Thanks @rlavarec-gpsw   I have change according to your suggestion.

**Review by rlavarec-gpsw — APPROVED** [review]

File: libnodegl/src/backends/gl/glcontext_eagl.m [hunk 2]
```diff
@@ -293,7 +293,7 @@ static int eagl_make_current(struct glcontext *ctx, int current)
         ret = [EAGLContext setCurrentContext:nil];
     }
 
-    return ret;
+    return ret != FALSE ? 0: -1;
```
Comment (rlavarec-gpsw) [inline libnodegl/src/backends/gl/glcontext_eagl.m:296]: It seems it is more `NO` use in pure Objective C.

https://developer.apple.com/documentation/objectivec/objective-c_runtime/boolean_values

Comment (hsong-gpsw) [reply libnodegl/src/backends/gl/glcontext_eagl.m]: right, corrected.  Thanks

File: libnodegl/src/backends/gl/glcontext_eagl.m [hunk 3]
```diff
@@ -293,7 +293,7 @@ static int eagl_make_current(struct glcontext *ctx, int current)
         ret = [EAGLContext setCurrentContext:nil];
     }
 
-    return ret;
+    return ret != NO ? 0: -1;
```
Comment (rlavarec-gpsw) [inline libnodegl/src/backends/gl/glcontext_eagl.m:296]: Sorry I think about it overnight and we could write the function the other way, to do not bother avec FALSE or NO or false.

This is working in every case and simpler to read.
`ret` is not returning, so we do not need to use it as a name.

My new final proposition
```
    BOOL isOK;
    struct eagl_priv *eagl = ctx->priv_data;

    if (current) {
        isOK = [EAGLContext setCurrentContext:eagl->handle];
    } else {
        isOK = [EAGLContext setCurrentContext:nil];
    }

    return isOK ? 0: -1;
```

**Review by fallera-gpsw — APPROVED** [review]

**Pull request merged**